conans/client/cache/remote_registry.py

```diff
@@ -181,19 +214,25 @@ def get(self, remote_name):
 
     def __getitem__(self, remote_name):
         try:
-            return self._remotes[remote_name]
+            remote = self._remotes[remote_name]
+            if remote.disabled:
+                raise ConanException("Remote '%s' is disabled" % (remote_name))
+            else:
+                return remote
         except KeyError:
             raise NoRemoteAvailable("No remote '%s' defined in remotes" % (remote_name))
 
     def __delitem__(self, remote_name):
+        if remote_name in self._remotes and self._remotes[remote_name].disabled:
```

Comment:
we don't want to remove a remote that it is disabled? Why?

Reply:
I was trying to be as close as possible to the behaviour in case the remote does not exist...
So should we enable operations like remove, update and rename for remotes that are disabled ?